.github/workflows/run_integration_tests.yml

```diff
@@ -2,13 +2,406 @@ name: Run Integration Tests
+            'Logging.MetricsAndForwarding.log4net', 'Logging.MetricsAndForwarding.MicrosoftLogging', 'Logging.MetricsAndForwarding.Serilog', 'Logging.MetricsAndForwarding.NLog', 'ReJit.NetFramework', 'RemoteServiceFixtures', 'RestSharp', \
+            'WCF.Client.IIS.ASPDisabled', 'WCF.Client.IIS.ASPEnabled', 'WCF.Client.Self', 'WCF.Service.IIS.ASPDisabled', 'WCF.Service.IIS.ASPEnabled', 'WCF.Service.Self', 'RequestHeadersCapture.WCF', 'RequestHeadersCapture.Owin', \
+            'RequestHeadersCapture.AspNetCore', 'RequestHeadersCapture.Asp35', 'RequestHeadersCapture.EnvironmentVariables', 'RequestHandling', 'CodeLevelMetrics' ]"
+          else
+            # Just use the supplied list of namespaces
+            namespaces="[ ${{ github.event.inputs.integration-test-namespaces }} ]"
+          fi
+          echo "::set-output name=matrix::$namespaces"
+        shell: bash
+
+      - id: configure_unbounded_tests_matrix
+        run: |
+          if [ "${{ github.event.inputs.unbounded-test-namespaces }}" == "ALL" ] ; then
+            # Use the full list of namespaces
+            namespaces="[ 'CosmosDB', 'Couchbase', 'MongoDB', 'Msmq', 'MsSql', 'MySql', 'NServiceBus', 'NServiceBus5', 'Oracle', 'Postgres', 'RabbitMq', 'Redis' ]"
+          else
+            # Just use the supplied list of namespaces
+            namespaces="[ ${{ github.event.inputs.unbounded-test-namespaces }} ]"
+          fi
+          echo "::set-output name=matrix::$namespaces"
+        shell: bash
+
+
+
+  run-integration-tests:
+    needs: [setup-matrices]
+    if: github.event.inputs.solutions == 'Both' || github.event.inputs.solutions == 'IntegrationTests'
+    name: Run IntegrationTests
+    runs-on: windows-2019 # TODO: make this a input so we can test different OSes?
+    strategy:
+      matrix:
+        namespace: ${{fromJSON(needs.setup-matrices.outputs.integration-tests-matrix)}}
+      fail-fast: false # we don't want one test failure in one namespace to kill the other runs
+
+    env:
+      integration_tests_shared_project: ${{ github.workspace }}/tests/Agent/IntegrationTests/Shared
+      xunit_console: ${{ github.workspace }}/build/Tools/XUnit-Console/xunit.console.exe
+      integration_tests_dll: ${{ github.workspace }}/tests/Agent/IntegrationTests/IntegrationTests/bin/Release/net462/NewRelic.Agent.IntegrationTests.dll
+      # Make this variable true to enable extra data-gathering and logging to help troubleshoot test failures, at the cost of additional time and resources
+      enhanced_logging: false
+      NR_DOTNET_TEST_SAVE_WORKING_DIRECTORY : 1
+      
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Setup .NET Core 2.2.103
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '2.2.103'
+
+      - name: Setup .NET Core 3.1.100
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: '3.1.100'
```

Comment:
Are you sure we need to install this? I think the server 2019 runner has 3.1.x installed already: https://github.com/actions/runner-images/blob/main/images/win/Windows2019-Readme.md#net-core-sdk